parsers/gpxfile.h

```diff
@@ -11,8 +11,8 @@ namespace gpxfile
 
 bool isFileDataSupported(const std::string& fileData);
 
-bool parseFileData(const std::string& fileData, Container<TrackPoint>& stream);
+bool parseFileData(const std::string& fileData, Container<TrackPoint>& stream, std::string& activityName, std::chrono::system_clock::time_point& activityStartTime);
 
-bool generateFileData(const Container<TrackPoint>& stream, std::string& fileData);
+bool generateFileData(const Container<TrackPoint>& stream, const std::string& activityName, std::chrono::system_clock::time_point activityStartTime, std::string& fileData);
```

Comment:
Move the Activity class from the activity_server module to the parser and use this instead of individual parameters and a track point list.